internal/Picker/Picker.js

```diff
@@ -0,0 +1,316 @@
+		 */
+		orientation: PropTypes.oneOf(['horizontal', 'vertical']),
+
+		/**
+		 * Allow the picker to only increment or decrement by a given value.
+		 *
+		 * A step of `2` would cause a picker to increment from 10 to 12 to 14, etc. It must evenly
+		 * divide into the range designated by `min` and `max`.
+		 *
+		 * @type {Number}
+		 * @default 1
+		 * @public
+		 */
+		step: PropTypes.number,
+
+		/**
+		 * Index of the selected child.
+		 *
+		 * @type {Number}
+		 * @default 0
+		 * @public
+		 */
+		value: PropTypes.number
+	},
+
+	defaultProps: {
+		orientation: 'vertical',
+		step: 1,
+		value: 0
+	},
+
+	styles: {
+		css,
+		className: 'picker',
+		publicClassNames: true
+	},
+
+	handlers: {
+		handleDecrement: decrement,
+		handleFlick: handle(
+			forEventProp('direction', 'vertical'),
+			// ignore "slow" flicks by filtering out velocity below a threshold
+			oneOf(
+				[({velocityY}) => velocityY < 0, increment],
+				[({velocityY}) => velocityY > 0, decrement]
+			)
+		),
+		handleIncrement: increment
+	},
+
+	computed: {
+		activeClassName: ({styler}) => styler.join('active', 'item'),
+		className: ({orientation, styler}) => styler.append(orientation),
+		decrementAriaLabel: ({'aria-valuetext': valueText, children: values, decrementAriaLabel = $L('previous item'), value}) => {
+			if (Array.isArray(values)) {
+				return `${valueText != null ? valueText : values[value]} ${decrementAriaLabel}`;
+			} else {
+				return `${valueText != null ? valueText : value} ${decrementAriaLabel}`;
+			}
+
+		},
+		valueId: ({id}) => `${id}_value`,
+		incrementAriaLabel: ({'aria-valuetext': valueText, children: values, incrementAriaLabel = $L('next item'), value}) => {
+			if (Array.isArray(values)) {
+				return `${valueText != null ? valueText : values[value]} ${incrementAriaLabel}`;
+			} else {
+				return `${valueText != null ? valueText : value} ${incrementAriaLabel}`;
+			}
+		},
+		currentValueText: ({'aria-valuetext': valueText, children: values, value}) => {
+			if (Array.isArray(values)) {
+				return `${typeof valueText !== 'undefined' ? valueText : values[value]}`;
+			} else {
+				return `${typeof valueText !== 'undefined' ? valueText : value}`;
+			}
+		}
+	},
+
+	render: (props) => {
+		const {
+			activeClassName,
+			children: values,
+			currentValueText,
+			decrementAriaLabel,
+			handleDecrement,
+			handleFlick,
+			handleIncrement,
+			incrementAriaLabel,
+			min,
+			max,
+			step,
+			value,
+			valueId,
+			...rest
+		} = props;
+		const isFirst = value <= min;
+		const isLast = value >= max;
+
+		return (
+			<PickerRoot {...rest} onFlick={handleFlick}>
+				<PickerButtonItem
+					aria-controls={valueId}
+					aria-disabled={isFirst}
+					aria-label={decrementAriaLabel}
+					className={css.itemDecrement}
+					onClick={handleDecrement}
+					disabled={isFirst}
+				>
+					<div className={css.label}>
+						{isFirst ? '' :  clamp(min, max, Array.isArray(values) ? values[value - step] : value - step)}
```

Comment:
The `clamp(min, max, Array.isArray(values)` log is used twice and i defined in JSX. It would be better to define it as a variable out of JSX, but in render function.

Reply:
ok, i will define currentValue, decrementValue and incrementValue inside render()